### PR TITLE
Add PHPUnit 4.8 to require-dev

### PR DESCRIPTION
**Pull request by shaunbramley**
As mentioned in reactphp/react#354

Added phpunit 4.8 to require-dev section of composer.json
Modified travisci config  to perform a composer selfupdate and to use local phpunit.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,11 @@ php:
...
 
+before_install:
+  - composer self-update
+
 install:
   - composer install --prefer-source --no-interaction
 
 script:
-  - phpunit --coverage-text
+  - ./vendor/binphpunit -v --coverage-text
```
Comment (jsor) [inline .travis.yml]: Missing slash, should be `./vendor/bin/phpunit`

File: .travis.yml [hunk 2]
```diff
@@ -10,8 +10,11 @@ php:
 
 sudo: false
 
+before_install:
+  - composer self-update
```
Comment (WyriHaximus) [inline .travis.yml]: There is no need for this anymore as [travis already do it here](https://travis-ci.org/reactphp/socket/jobs/190057478#L146) while your update call runs [here](https://travis-ci.org/reactphp/socket/jobs/190057478#L161).

**Comment by shaunbramley** [comment]
pushed changes as requested.

File: .travis.yml [hunk 3]
```diff
@@ -14,4 +14,4 @@ install:
   - composer install --prefer-source --no-interaction
 
 script:
-  - phpunit --coverage-text
+  - ./vendor/bin/phpunit -v --coverage-text
```
Comment (clue) [inline .travis.yml]: What's the motivation for adding the verbose flag?

**Comment by shaunbramley** [comment]
To be quite honest. I saw it in a couple of the other repos. I've seen variety of differences in how each Travis.yml file is setup. I've been contemplating posting a "bug report" on the meta repo to begin a discussion about a standardized (where possible) configuration. 

For example:
- store phpunit analysis in clover.xml format for post analysis?
- test against lowest deps ?
- test against highest deps?
- test against PHP 7.1?


Cheers,
Shaun

**Comment by clue** [comment]
> discussion about a standardized (where possible) configuration

I think this makes perfect sense :+1: 

In the meantime, I would suggest reverting this (unrelated and undocumented) change so we can get this in quicker :+1: Also, may I ask you to sqash/force-push your changes so this is a single commit only? Thanks!

**Comment by shaunbramley** [comment]
@clue reverted / squashed / force-pushed as requested.

**Comment by clue** [comment]
Thanks! The build currently fails because this is based on on older version of the master branch. A rebase should help :+1:

**Comment by shaunbramley** [comment]
@clue rebased off of upstream master as requested

**Pull request merged**